test/js_test.go

```diff
@@ -1421,6 +1421,106 @@ Loop:
 	})
 }
 
+func TestJetStreamPushFlowControl_SubscribeAsyncAndChannel(t *testing.T) {
+	s := RunBasicJetStreamServer()
+	defer s.Shutdown()
+
+	if config := s.JetStreamConfig(); config != nil {
+		defer os.RemoveAll(config.StoreDir)
+	}
+
+	errCh := make(chan error)
+	errHandler := nats.ErrorHandler(func(c *nats.Conn, sub *nats.Subscription, err error) {
+		t.Logf("WARN: %s : %+v", err, sub)
+		errCh <- err
+	})
+	nc, err := nats.Connect(s.ClientURL(), errHandler)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	defer nc.Close()
+
+	const totalMsgs = 10000
```

Comment:
To hit the limit locally, I need to make this a higher value like more than 30000 msgs here, but that will fail in Travis since environment is slower.